routing/router_test.go

```diff
@@ -2922,16 +2922,16 @@ func TestSendToRouteStructuredError(t *testing.T) {
 	}
 
 	// We'll modify the SendToSwitch method so that it simulates a failed
-	// payment with an error originating from the first hop of the route.
-	// The unsigned channel update is attached to the failure message.
+	// payment with an error originating from the final hop in the route.
+	finalHopIndex := len(hops)
```

Comment:
I think there's some value here, in just adding this as a new test case instead.